File: game/scripts/vscripts/modifiers/generic/modifier_tower_idle.lua
modifier_tower_idle = class({})

function modifier_tower_idle:IsHidden()                  return true     end
function modifier_tower_idle:IsDebuff()                  return false    end
function modifier_tower_idle:IsPurgable()                return false    end
function modifier_tower_idle:StatusEffectPriority()      return 1.0      end

function modifier_tower_idle:DeclareFunctions() 
    return {
        MODIFIER_PROPERTY_OVERRIDE_ANIMATION,
        MODIFIER_PROPERTY_OVERRIDE_ANIMATION_RATE,
    }
end

function modifier_tower_idle:GetOverrideAnimation(params)          return ACT_DOTA_CUSTOM_TOWER_IDLE     end
function modifier_tower_idle:GetOverrideAnimationRate(params)      return 1.0              end

File: src/vscripts/modifiers/modifier_tower_idle.ts
import { CustomModifier } from "../abilities/framework/custom_modifier";
import { registerModifier } from "../lib/dota_ts_adapter";

@registerModifier({ customNameForI18n: "modifier_tower_idle" })
export class ModifierTowerIdle extends CustomModifier<undefined> {
    IsHidden() {
        return true;
    }

    IsDebuff() {
        return false;
    }

    IsPurgable() {
        return false;
    }

    StatusEffectPriority() {
        return 1.0;
    }

    DeclareFunctions() {
        return [ModifierFunction.OVERRIDE_ANIMATION, ModifierFunction.OVERRIDE_ANIMATION_RATE];
    }

    GetOverrideAnimation() {
        return GameActivity.DOTA_CUSTOM_TOWER_IDLE;
    }

    GetOverrideAnimationRate() {
        return 1.0;
    }
}
